test/spmd/test_spmd_lowering_context.py

```diff
@@ -18,6 +21,87 @@ class TestSPMDLoweringContext(test_xla_sharding_base.XlaShardingTest):
   def setUpClass(cls):
     super().setUpClass()
 
+  def _get_computation_hlo_txt(self, ctx):
+    hlo = ctx.hlo()
+    comp = xb.computation_from_module_proto("my_custom_comp", hlo)
+    return xb.get_computation_hlo(comp)
+
+  def test_basic(self):
+    save_file = os.getenv('XLA_SAVE_TENSORS_FILE')
+    save_format = os.getenv('XLA_SAVE_TENSORS_FMT')
+    assert save_file, "This test should be run with XLA_SAVE_TENSORS_FILE"
+    save_file += '.0'  # Identify a single device
+    assert save_format == 'hlo', "This test should be run with XLA_SAVE_TENSORS_FMT=hlo"
+
+    model_axis = min(8, self.n_devices)
```

Comment:
I see, nothing specifically. `self.n_devices // 2` or 2 could be troubling if a single device runs the test, which it can. These SPMD aren't skipped if so by design. I just handled the single device corner case in `model_axis`, but should suffice without hardcoding a specific number.